src/main/java/examples/Examples.java

```diff
@@ -410,4 +410,24 @@ public void example12(MongoService mongoService) {
 
   }
 
+  public void example13_0(MongoService mongoService) {
+
+    JsonObject document = new JsonObject().put("title", "The Hobbit")
+        .put("publicationDate", new JsonObject().put("$date", -1018659600000L /* 21/9/1937 */));
```

Comment:
As it stands right now the Vert-x mongo client support is a pretty straight binding to the async Java driver (which is a good thing IMO).  Given that this driver provides a codec which directly handles java.util.Date -- see `org.bson.codecs.DateCodec`, I'm not sure that eliminating that is the right way to go.  I'd prefer to see things remain as close to the Java driver as possible (with the one wrinkle of serializing to/from JsonObject/JsonArray since that is Vert.x's native Java form).

Also the `{"$date": "<date>"}` form is referred to as the the strict form while the shell mode for date is simply `new Date(<date)`.  Pendantic I know, but...